microsetta_public_api/models/_taxonomy.py

```diff
@@ -206,3 +219,60 @@ def get_group(self, ids: Iterable[str], name: str = None) -> GroupTaxonomy:
                              feature_variances=list(feature_variances),
                              feature_ranks=None,
                              )
+
+    def presence_data_table(self, ids: Iterable[str]) -> DataTable:
+        table = self._table.filter(set(ids), inplace=False).remove_empty()
+        features = table.ids(axis='observation')
+
+        entries = list()
+        for vec, sample_id, _ in table.iter(dense=False):
+            for feature_idx, val in zip(vec.indices, vec.data):
+                entries.append({
+                    **{'sampleId': sample_id,
+                       'relativeAbundance': val},
+                    **self._formatted_taxa_names[features[feature_idx]],
```

Comment:
I'm not sure I understand what this does. Possible to rewrite it? I think the following is consistent

```python
entry = {'sampleId': sample_id, 'relativeAbundance', val}
entry.update(self._formatted_taxa_names[features[feature_idx]])
entries.append(entry)
```